CHANGELOG.md

```diff
@@ -10,9 +10,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Inserted words into crossword grid
 
 ### Changed
-- Some improvement ...
-- Updated ...
-- Refactored ...
+- Updated word filling algorithm to check whether valid words can also be filled in the lines 
+intersecting with the current line in order to reduce the frequency of unnecessary backtracking
+- Refactored Grid class (PR #4 to #14)
```

Comment:
```suggestion
- Added .. (a new feature,  a new functionality, or component. It tells the user that something new is available)
- Fixed/Resolved .. (an issue, bug, or problem has been resolved. Confirmation that the issue has been addressed)
- Updated.. (something existing has been modified or brought up to date.)
- Removed .. (something has been taken out (a deprecated feature, an old dependency, or unnecessary functionality). It indicates that something is no longer available)
- Improved .. (enhancements or optimizations that make a feature or functionality better, whether in terms of performance, usability, or security)

### Changed
- Update word filling to check if valid words can fill the intersecting lines to reduce back tracking frequency (PR#)
- Updated grid class (for what?) (PR #4 to #14)
```

Reply:
1. Keep the Added, Fixed, Update, Removed, Improved as sample. These are the top five imperative words change log entry begins with.
2. Practice on using these words for change log entry because they effectively communicate the nature of the changes in a concise and clear manner.
3. Use specific terms that directly communicate the impact or benefit to the user rather than the internal process (like "refactoring"). The focus should be on what has improved or changed from the user's perspective.

Reply:
Since refactoring doesn't change anything from the user's perspective I will remove it, and will rephrase changes to the word filling algorithm to be more user-focused.